packages/ra-ui-materialui/src/layout/Notification.tsx

```diff
@@ -51,6 +62,10 @@ export const Notification = (props: NotificationProps) => {
             // Close an active snack when a new one is added
             setOpen(false);
         }
+
+        return () => {
+            window.removeEventListener('beforeunload', beforeunload);
```

Comment:
remove the event listener only if you added it! You must enclose this in the same condition as line 53